tests/kolainst/destructive/finalization.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -xeuo pipefail
+
+. ${KOLA_EXT_DATA}/libinsttest.sh
+
+require_writable_sysroot
+prepare_tmpdir
+
+case "${AUTOPKGTEST_REBOOT_MARK:-}" in
+  "")
+  # Need to disable gpg verification for test builds
+  sed -i -e 's,gpg-verify=true,gpg-verify=false,' /etc/ostree/remotes.d/*.conf
+  # xref https://github.com/coreos/coreos-assembler/pull/2814
+  systemctl mask --now zincati
+
+  # Test the deploy --stage functionality; first, we stage a deployment
+  # reboot, and validate that it worked.
+  # Write staged-deploy commit
+  cd /ostree/repo/tmp
+  ostree checkout -H ${host_commit} t
+
+  # Create a synthetic commit for upgrade
+  ostree commit --no-bindings --parent="${host_commit}" -b staged-deploy -I --consume t
```

Comment:
Minor: I think we could combine these two and avoid the checkout with `--tree=ref=${host_commit}`, which we use in other tests.